pkg/cloudproduct/gke/gke_test.go

```diff
@@ -222,6 +224,52 @@ func TestPodSeccompUnconfined(t *testing.T) {
 	}
 }
 
+func TestSetPassthroughLabel(t *testing.T) {
+	for name, tc := range map[string]struct {
+		pod      *corev1.Pod
+		wantPod  *corev1.Pod
+		ports    []agonesv1.GameServerPort
+		features string
+	}{
+		"gameserver with with Passthrough port policy adds label to pod": {
+			features: fmt.Sprintf("%s=true", runtime.FeatureAutopilotPassthroughPort),
+
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{},
+					Labels:      map[string]string{},
+				},
+			},
+			ports: []agonesv1.GameServerPort{
+				{
+					Name:          "awesome-udp",
+					PortPolicy:    agonesv1.Passthrough,
+					ContainerPort: 1234,
+					Protocol:      corev1.ProtocolUDP,
+				},
+			},
+			wantPod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{},
+					Labels: map[string]string{
+						agonesv1.GameServerPortPolicyPodLabel: "autopilot-passthrough",
+					},
+				},
+			},
+		},
```

Comment:
add negative tests for:
* feature gate disabled
* no passthrough policy
just to verify it's not modifying anything in those cases. (I thikn it would trip other tests but might as well)

Reply:
Sure, done!